src/ansys/fluent/visualization/plotter/plotter_windows_manager.py

```diff
@@ -130,19 +131,17 @@ def _show_plot(self):
         self.plotter.show()
 
     # private methods
-    def _get_plotter(self):
-        import ansys.fluent.visualization as pyviz
+    def _get_plotter(self, plotter=None):
+        if plotter is None:
+            import ansys.fluent.visualization as pyviz
 
-        if pyviz.PLOTTER == "matplotlib":
-            from ansys.fluent.visualization.plotter.matplotlib.plotter_defns import (
-                Plotter,
-            )
-        elif pyviz.PLOTTER == "plotly":
-            from ansys.fluent.visualization.plotter.plotly.plotter_defns import Plotter
-        else:
-            from ansys.fluent.visualization.plotter.pyvista.plotter_defns import Plotter
+            plotter = pyviz.PLOTTER
+        try:
+            plotter = get_visualizer(plotter)
+        except KeyError as ex:
+            raise KeyError("Please register custom plotter before using it.") from ex
```

Comment:
@prmukherj I don't follow this logic. Please could we go through it in the meeting?

Reply:
Done.